ownCloud/Client/Actions/EditDocumentViewController.swift

```diff
@@ -95,26 +96,45 @@ class EditDocumentViewController: QLPreviewController, Themeable {
 		}
 	}
 
-	override func viewDidAppear(_ animated: Bool) {
-		super.viewDidAppear(animated)
-		enableEditingMode()
+	required init?(coder aDecoder: NSCoder) {
+		fatalError("init(coder:) has not been implemented")
+	}
+
+	deinit {
+		Theme.shared.unregister(client: self)
+		timer = nil
+	}
+
+	override func viewDidLoad() {
+		super.viewDidLoad()
+
+		let queue = DispatchQueue(label: "com.owncloud.edit-document-queue")
+		timer = DispatchSource.makeTimerSource(queue: queue)
```

Comment:
@hosy timer always smells after race conditions.. Wouldn't it be enough to put the code within delayed block to `viewDidAppear()` for instance?

Reply:
We had this implementation before and it does not work, because some items are available immediately and some items needs more seconds to load.